packages/js/src/utils/package-json/update-package-json.ts

```diff
@@ -44,45 +46,68 @@ export function updatePackageJson(
   target: ProjectGraphProjectNode,
   dependencies: DependentBuildableProjectNode[]
 ): void {
-  const pathToPackageJson = join(
-    context.root,
-    options.projectRoot,
-    'package.json'
+  const packageJson = createPackageJson(
+    context.projectName,
+    context.projectGraph,
+    {
+      root: context.root,
+      // By default we remove devDependencies since this is a production build.
+      isProduction: true,
+    }
   );
-
-  const packageJson = fileExists(pathToPackageJson)
-    ? readJsonFile(pathToPackageJson)
-    : { name: context.projectName };
-
+  // make custom modifications to package.json
+  updatePackageJsonContent(packageJson, options);
   if (options.excludeLibsInPackageJson) {
     dependencies = dependencies.filter((dep) => dep.node.type !== 'lib');
   }
+  if (options.updateBuildableProjectDepsInPackageJson) {
+    addMissingLibDependencies(packageJson, context, dependencies);
+  }
 
-  writeJsonFile(
-    `${options.outputPath}/package.json`,
-    getUpdatedPackageJsonContent(packageJson, options)
-  );
+  writeJsonFile(`${options.outputPath}/package.json`, packageJson);
+  const lockFile = createLockFile(packageJson);
+  writeFileSync(`${options.outputPath}/${getLockFileName()}`, lockFile, {
+    encoding: 'utf-8',
+  });
+}
 
-  if (
-    dependencies.length > 0 &&
-    options.updateBuildableProjectDepsInPackageJson
-  ) {
-    updateBuildableProjectPackageJsonDependencies(
-      context.root,
-      context.projectName,
-      context.targetName,
-      context.configurationName,
-      target,
-      dependencies,
-      options.buildableProjectDepsInPackageJsonType
-    );
-  }
+function addMissingLibDependencies(
+  packageJson: PackageJson,
+  { projectName, targetName, configurationName, root }: ExecutorContext,
+  dependencies: DependentBuildableProjectNode[]
+) {
+  dependencies.forEach((entry) => {
+    if (!isNpmProject(entry.node)) {
+      const packageName = entry.name;
```

Comment:
This is copied over and simplified from `updateBuildableProjectPackageJsonDependencies`